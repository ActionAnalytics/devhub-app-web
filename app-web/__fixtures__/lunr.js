export const LUNR_SEARCH_RESULTS = [
  {
    title: 'FOO',
    collectionName: 'Design System',
    description: 'This is the design system',
    author: 'bcgov',
    resourceType: 'Components',
    id: '1',
    collectionId: 'collection-1',
  },
  {
    title: 'FOO2',
    collectionName: 'Design System',
    description: 'This is the design system foo2',
    author: 'bcgov',
    resourceType: 'Components',
    id: '2',
    collectionId: 'collection-1',
  },
  {
    title: 'FOO3',
    collectionName: 'Design System',
    description: 'This is the design system foo3',
    author: 'bcgov',
    resourceType: 'Components',
    id: '3',
    collectionId: 'collection-1',
  },
  {
    title: 'FOO4',
    collectionName: 'Design System',
    description: 'This is the design system',
    author: 'bcgov',
    resourceType: 'Components',
    id: '4',
    collectionId: 'collection-1',
  },
  {
    title: 'BAZ1',
    collectionName: 'Devhub',
    description: 'This is the devhub',
    author: 'bcgov',
    resourceType: 'Documentation',
    id: '5',
    collectionId: 'collection-2',
  },
  {
    title: 'BAZ2',
    collectionName: 'Devhub',
    description: 'This is the devhub again',
    author: 'bcgov',
    resourceType: 'Documentation',
    id: '6',
    collectionId: 'collection-2',
  },
];

