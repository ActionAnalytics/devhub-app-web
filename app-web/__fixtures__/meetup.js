/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Derek Siemens
*/

// Meetup sample data

const MEETUP_EVENT = {
  id: 'a5293451-4f1d-5518-8099-e2f2bcb25891',
  internal: {
    content:
      '{"created":1561565461000,"duration":3600000,"id":"262643690","name":"Anything but Jenkins, long live Jenkins ","rsvp_limit":41,"date_in_series_pattern":false,"status":"past","time":1562097600000,"local_date":"2019-07-02","local_time":"13:00","updated":1562107379000,"utc_offset":-25200000,"waitlist_count":0,"yes_rsvp_count":31,"venue":{"id":25859211,"name":"BC Government Continuous Service Improvement Lab - Grand Central Station (3rd Floor Common Area)","lat":48.42424011230469,"lon":-123.36581420898438,"repinned":true,"address_1":"3rd Floor 1012 Douglas St.","city":"Victoria","country":"ca","localized_country_name":"Canada","zip":"V8W 2C3","state":"bc"},"group":{"created":1526511874000,"name":"BC DevOps Commons","id":28512032,"join_mode":"open","lat":48.43000030517578,"lon":-123.36000061035156,"urlname":"DevOps-Commons","who":"Members","localized_location":"Victoria, BC","state":"BC","country":"ca","region":"en_US","timezone":"Canada/Pacific"},"link":"https://www.meetup.com/DevOps-Commons/events/262643690/","description":"<p>We love Jenkins, but thereâ€™s got to be a better way... join pipeline experts as we demo and explore Continuous Integration and Continuous Delivery tools and processes.</p> <p>Click here to join remotely: <a href=\\"https://meet.google.com/cmd-fxui-myk\\" class=\\"linkified\\">https://meet.google.com/cmd-fxui-myk</a></p> ","visibility":"public_limited"}',
    contentDigest: 'bc323e4419c8a3eff7a41dea2cfaa4b2',
    description: 'DEMO MEETUP DATA - DEVELOPER ONLY ',
    ignoreType: null,
    mediaType: null,
    owner: 'gatsby-source-meetup',
    type: 'MeetupEvent',
  },
  created: 1561565461000,
  duration: 3600000,
  name: 'Anything but Jenkins, long live Jenkins ',
  rsvp_limit: 41,
  date_in_series_pattern: false,
  status: 'past',
  time: 1562097600000,
  local_date: '2019-07-02',
  local_time: '13:00',
  updated: 1562107379000,
  utc_offset: -25200000,
  waitlist_count: 0,
  yes_rsvp_count: 31,
  venue: {
    id: 25859211,
    name:
      'BC Government Continuous Service Improvement Lab - Grand Central Station (3rd Floor Common Area)',
    lat: 48.42424011230469,
    lon: -123.36581420898438,
    repinned: true,
    address_1: '3rd Floor 1012 Douglas St.',
    city: 'Victoria',
    country: 'ca',
    localized_country_name: 'Canada',
    zip: 'V8W 2C3',
    state: 'bc',
    address_2: null,
    phone: null,
  },
  group: {
    created: 1526511874000,
    name: 'BC DevOps Commons',
    id: 28512032,
    join_mode: 'open',
    lat: 48.43000030517578,
    lon: -123.36000061035156,
    urlname: 'DevOps-Commons',
    who: 'Members',
    localized_location: 'Victoria, BC',
    state: 'BC',
    country: 'ca',
    region: 'en_US',
    timezone: 'Canada/Pacific',
  },
  link: 'https://www.meetup.com/DevOps-Commons/events/262643690/',
  description: 'DEMO MEETUP',
  visibility: 'public_limited',
  fields: {
    name: 'DEMO MEETUP DATA - DEVELOPER ONLY ',
    description: 'DEMO MEETUP DATA - DEVELOPER ONLY ',
    link: 'https://www.meetup.com/DevOps-Commons/events/262643690/',
    location: '3rd Floor 1012 Douglas St.',
  },
  how_to_find_us: null,
  pro_is_email_shared: null,
  rsvp_close_offset: null,
  fee: null,
  manual_attendance_count: null,
};

module.exports = {
  MEETUP_EVENT,
};
