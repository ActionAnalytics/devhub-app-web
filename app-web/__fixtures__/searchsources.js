/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
// run time apollo query fixtures
export const GITHUB = [
  {
    type: 'github',
    id: '101',
    typePayload: JSON.stringify({
      id: '1',
      __typename: 'Github Issue',
      fields: {
        resourceType: 'Github Issue',
        title: 'Issue',
        description: 'Foo',
        personas: ['Developer'],
      },
    }),
  },
  {
    type: 'github',
    id: '102',
    typePayload: JSON.stringify({
      id: '2',
      __typename: 'Github Issue',
      fields: {
        resourceType: 'Github Issue',
        title: 'Issue',
        description: 'Foo',
        personas: ['Developer'],
      },
    }),
  },
];
export const ROCKET_CHAT = [
  {
    type: 'rocketchat',
    id: '1',
    typePayload: JSON.stringify({
      id: '1',
      roomId: '2',
      author: 'apple',
      time: '2019-02-12',
      message: 'apples are clearly superior',
    }),
  },
  {
    type: 'rocketchat',
    id: '2',
    typePayload: JSON.stringify({
      id: '2',
      roomId: '2',
      author: 'apple',
      time: '2019-02-12',
      message: 'apples are clearly superior',
    }),
  },
];
