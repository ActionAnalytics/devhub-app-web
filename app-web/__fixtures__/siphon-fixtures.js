const IDS = {
  DESIGN_SYSTEM: 'collection-1',
  DEVHUB: 'collection-2',
};

// SIPHON NODES
export const DESIGN_SYSTEM_NODE_1 = {
  id: '1',
  parent: {
    id: IDS.DESIGN_SYSTEM,
  },
  _metadata: {
    position: [0, 0, 0],
  },
  attributes: {
    personas: ['Designer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Components',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DESIGN_SYSTEM_NODE_2 = {
  id: '2',
  parent: {
    id: IDS.DESIGN_SYSTEM,
  },
  _metadata: {
    position: [0, 0, 1],
  },
  attributes: {
    personas: ['Designer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Components',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DESIGN_SYSTEM_NODE_3 = {
  id: '3',
  parent: {
    id: IDS.DESIGN_SYSTEM,
  },
  _metadata: {
    position: [0, 0, 2],
  },
  attributes: {
    personas: ['Designer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Components',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DESIGN_SYSTEM_NODE_4 = {
  id: '4',
  parent: {
    id: IDS.DESIGN_SYSTEM,
  },
  _metadata: {
    position: [0, 0, 3],
  },
  attributes: {
    personas: ['Developer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Components',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DEVHUB_NODE_1 = {
  id: '5',
  parent: {
    id: IDS.DEVHUB,
  },
  _metadata: {
    position: [1, 0, 0],
  },
  attributes: {
    personas: ['Developer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Documentation',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DEVHUB_NODE_2 = {
  id: '6',
  parent: {
    id: IDS.DEVHUB,
  },
  _metadata: {
    position: [1, 0, 1],
  },
  attributes: {
    personas: ['Developer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Documentation',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DESIGN_SYSTEM_NODES = [
  DESIGN_SYSTEM_NODE_1,
  DESIGN_SYSTEM_NODE_2,
  DESIGN_SYSTEM_NODE_3,
  DESIGN_SYSTEM_NODE_4,
];

export const DEVHUB_NODES = [DEVHUB_NODE_1, DEVHUB_NODE_2];

// COLLECTIONS
export const DESIGN_SYSTEM_COLLECTION = {
  id: IDS.DESIGN_SYSTEM,
  type: 'default',
  title: 'Design System',
  description: 'baz',
  nodes: DESIGN_SYSTEM_NODES,
};

export const DEVHUB_COLLECTION = {
  id: IDS.DEVHUB,
  type: 'default',
  title: 'Devhub',
  description: 'baz',
  nodes: DEVHUB_NODES,
};

export const SIPHON_NODES = DESIGN_SYSTEM_NODES.concat(DEVHUB_NODES);

export const COLLECTIONS = [DESIGN_SYSTEM_COLLECTION, DEVHUB_COLLECTION];
