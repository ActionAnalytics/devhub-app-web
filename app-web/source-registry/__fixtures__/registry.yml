sources:
  - name: DevHub
    sourceType: 'github'
    sourceProperties:
      url: https://github.com/bcgov/devhub-app-web/
      owner: bcgov
      repo: 'devhub-app-web'
    resourceType: 'Repositories'
    attributes:
      labels:
        - Repository
      persona: 'Developer'
