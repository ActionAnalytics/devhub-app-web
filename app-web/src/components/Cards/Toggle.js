import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Cards.module.css';
import Button from '../UI/Button/Button';

/**
 * Toggle component
 * This component has an arrow button to toggle the Cards to show all or collapse
 * @prop cardComponents containing each card
 * @prop cardLimits specify how many cards to shows as inital view
 */
class Toggle extends Component {
  state = {
    toggled: false,
  };

  toggledHandler = toggled => this.setState({ toggled });

  render() {
    const toggleIcon = this.state.toggled ? (
      <Button
        type="link"
        clicked={() => {
          this.toggledHandler(false);
        }}
      >
        Collapse
      </Button>
    ) : (
      <Button
        type="link"
        clicked={() => {
          this.toggledHandler(true);
        }}
      >
        Show All
      </Button>
    );
    const { cardComponents, cardLimits } = this.props;
    const cardShow = this.state.toggled ? cardComponents.length : cardLimits;
    const toggle =
      cardComponents.length > cardLimits ? (
        <div className={styles.ToggleButton}>{toggleIcon}</div>
      ) : null;

    return (
      <div>
        <div className={styles.Cards}>{cardComponents.slice(0, cardShow)}</div>
        {toggle}
      </div>
    );
  }
}

Toggle.propTypes = {
  cardComponents: PropTypes.array.isRequired,
  cardLimits: PropTypes.number.isRequired,
};

export default Toggle;
