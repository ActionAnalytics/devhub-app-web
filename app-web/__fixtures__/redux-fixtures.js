/*
Copyright 2018 Province of British Columbia
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 
   http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
Created by Patrick Simonian
*/
import * as actions from '../src/store/actions/actionTypes';
import { LUNR_SEARCH_RESULTS } from './lunr';
export const COLLECTIONS = [
  {
    name: 'Design System',
    type: 'source',
    nodes: [
      {
        _metadata: {
          position: [0, 0, 0],
        },
        attributes: {
          personas: ['Designer'],
        },
        source: {
          name: 'design-system',
          displayName: 'Design System',
          sourcePath: 'https://github.com/bcgov/design-system/',
          type: 'github',
        },
        resource: {
          path: '/design-system/Z1bGIqu',
          type: 'Components',
          originalSource:
            'https://github.com/bcgov/design-system/blob/master/components/about/about.md',
        },
        unfurl: {
          type: 'markdown',
          label1: '',
          data1: '',
          label2: '',
          data2: '',
          image:
            'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
          title: 'About',
          description: 'What the Design System is and how it works.',
          author: 'ksingbeil',
        },
        childMarkdownRemark: {
          frontmatter: {
            pageOnly: false,
          },
        },
      },
      {
        _metadata: {
          position: [0, 0, 1],
        },
        attributes: {
          personas: ['Developer'],
        },
        source: {
          name: 'design-system',
          displayName: 'Design System',
          sourcePath: 'https://github.com/bcgov/design-system/',
          type: 'github',
        },
        resource: {
          path: '/design-system/Z1bGIqu',
          type: 'Documentation',
          originalSource:
            'https://github.com/bcgov/design-system/blob/master/components/about/about.md',
        },
        unfurl: {
          type: 'markdown',
          label1: '',
          data1: '',
          label2: '',
          data2: '',
          image:
            'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
          title: 'About',
          description: 'What the Design System is and how it works.',
          author: 'ksingbeil',
        },
        childMarkdownRemark: {
          frontmatter: {
            pageOnly: false,
          },
        },
      },
    ],
  },
];

export const DEFAULT_FILTER_GROUPS = [
  {
    filterBy: 'attributes.personas',
    value: 'Designer',
    text: 'Designers',
    active: false,
    availableResources: 0,
    isFilterable: false,
    key: 'foo',
    title: 'For',
  },
  {
    filterBy: 'attributes.personas',
    value: 'Developer',
    text: 'Developers',
    active: true,
    availableResources: 0,
    isFilterable: false,
    key: 'bar',
    title: 'For',
  },
  {
    filterBy: 'attributes.personas',
    value: 'Product Owner',
    text: 'Product Owners',
    active: false,
    availableResources: 0,
    isFilterable: false,
    key: 'baz',
    title: 'For',
  },
];

export const ACTIONS = {
  ADD_FILTER: {
    type: actions.ADD_FILTER,
    payload: {
      key: 'foo',
    },
  },
  REMOVE_FILTER: {
    type: actions.REMOVE_FILTER,
    payload: {
      key: 'bar',
    },
  },
  LOAD_SIPHON_COLLECTIONS: {
    type: actions.LOAD_SIPHON_COLLECTIONS,
    payload: {
      nodes: COLLECTIONS,
    },
  },
  TOGGLE_FILTER_GROUP: {
    type: actions.TOGGLE_FILTER_GROUP,
    payload: {
      key: 'foo',
    },
  },
  FILTER_SIPHON_NODES: {
    type: actions.FILTER_SIPHON_NODES,
    payload: {
      filteredBy: 'resource.type',
      value: 'Components',
    },
  },
  FILTER_SIPHON_NODES_BY_ALL: {
    type: actions.FILTER_SIPHON_NODES,
    payload: {
      filteredBy: 'resource.type',
      value: 'All',
    },
  },
  FILTER_SIPHON_NODES_BY_LIST: {
    type: actions.FILTER_SIPHON_NODES_BY_FILTER_LIST,
  },
  SELECT_FILTER_OPTION: {
    type: actions.SET_SELECTED_FILTER_OPTION,
    payload: {
      foo: 'bar',
    },
  },
  TOGGLE_MAIN_NAVIGATION_ON: {
    type: actions.TOGGLE_MAIN_NAVIGATION,
    payload: {
      toggled: true,
    },
  },
  TOGGLE_MAIN_NAVIGATION_OFF: {
    type: actions.TOGGLE_MAIN_NAVIGATION,
    payload: {
      toggled: false,
    },
  },
  AUTHENTICATE_SUCCESS: {
    type: actions.AUTHENTICATE_SUCCESS,
  },
  AUTHENTICATE_FAILED: {
    type: actions.AUTHENTICATE_FAILED,
  },
  SET_SEARCH_RESULTS: {
    type: actions.SET_SEARCH_RESULTS,
    payload: {
      searchResults: LUNR_SEARCH_RESULTS,
    },
  },
};

export const INITIAL_STATES = {
  SIPHON: {
    collections: [],
    primaryFilteredNodes: [],
    secondaryFilteredNodes: [],
    filters: DEFAULT_FILTER_GROUPS,
    groupBy: null,
    loading: false,
    error: false,
    messages: [],
  },
  UI: {
    selectedFilterOption: null,
    mainNavigationToggled: false,
    welcomePanelWasViewed: false,
  },
  AUTH: {
    isAuthenticated: false,
    accessToken: null,
    idToken: null,
    error: false,
    messages: [],
  },
};
