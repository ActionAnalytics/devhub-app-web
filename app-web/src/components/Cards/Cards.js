import React from 'react';
import shortid from 'shortid';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGithub } from '@fortawesome/fontawesome-free-brands';
import styles from './Cards.module.css';
import DefaultCard from './Card/DefaultCard';
import DocumentCard from './Card/DocumentCard';
import RepositoryCard from './Card/RepositoryCard';
import SelfServiceCard from './Card/SelfServiceCard';
import ComponentCard from './Card/ComponentCard';
import Link from '../UI/Link/Link';
import Toggle from './Toggle';
import { ARIA_LABEL_REPO } from '../../constants/ariaLabels';
import { LARGE_SCREEN_LIMIT, SMALL_SCREEN_LIMIT, RESOURCE_TYPES } from '../../constants/ui';

const Cards = ({ topic, sourcePath, cards }) => {
  const cardComponents = cards.map(c => {
    switch (c.resourceType) {
      case RESOURCE_TYPES.RESPOSITORIES:
        return <RepositoryCard {...c} key={shortid.generate()} />;
      case RESOURCE_TYPES.SELF_SERVICE_TOOLS:
        return <SelfServiceCard {...c} key={shortid.generate()} />;
      case RESOURCE_TYPES.COMPONENTS:
        return <ComponentCard {...c} key={shortid.generate()} />;
      case RESOURCE_TYPES.DOCUMENTATION:
        return <DocumentCard {...c} key={shortid.generate()} />;
      default:
        return <DefaultCard {...c} key={shortid.generate()} />;
    }
  });
  const octoKat =
    sourcePath === '' ? null : (
      <Link to={sourcePath} aria-label={ARIA_LABEL_REPO}>
        <FontAwesomeIcon icon={faGithub} color="#242424" size="1x" />
      </Link>
    );
  return (
    <section className={styles.CardsContainer}>
      <div className={styles.TopicContainer}>
        <h1>{topic}</h1>
        {octoKat}
      </div>
      <div className={styles.LargeView}>
        <Toggle cardComponents={cardComponents} cardLimits={LARGE_SCREEN_LIMIT} />
      </div>
      <div className={styles.MobileView}>
        <Toggle cardComponents={cardComponents} cardLimits={SMALL_SCREEN_LIMIT} />
      </div>
    </section>
  );
};

Cards.defaultProps = {
  sourcePath: '',
};

Cards.propTypes = {
  cards: PropTypes.arrayOf(PropTypes.object).isRequired,
  sourcePath: PropTypes.string,
  topic: PropTypes.string.isRequired,
};

export default Cards;
