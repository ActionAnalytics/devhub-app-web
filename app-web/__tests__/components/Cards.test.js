import React from 'react';
import { shallow } from 'enzyme';
import Cards, { getIdealCardsLargeLimit } from '../../src/components/Cards/Cards';
import { COLLECTIONS } from '../../__fixtures__/siphon-fixtures';

describe('Card Component', () => {
  const CARDS = {
    DEFAULT: COLLECTIONS[0].nodes[0],
    REPOSITORY: COLLECTIONS[0].nodes[0],
    SELF_SERVICE: COLLECTIONS[0].nodes[0],
    COMPONENT: COLLECTIONS[0].nodes[0],
    DOCUMENT: COLLECTIONS[0].nodes[0],
  };

  it('matches snapshot', () => {
    const wrapper = shallow(<Cards cards={[CARDS.DEFAULT]} topic="topic" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('calculates toggle card limit correctly', () => {
    const MAX_CARD_LIMIT = 9;
    const CARDS_PER_ROW = 3;
    const NUM_CARDS1 = 8;
    const NUM_CARDS2 = 6;
    const NUM_CARDS3 = 2;
    const NUM_CARDS4 = 9;
    const NUM_CARDS5 = 10;
    const NUM_CARDS6 = 0;
    const NUM_CARDS7 = 4;

    expect(getIdealCardsLargeLimit(MAX_CARD_LIMIT, NUM_CARDS1, CARDS_PER_ROW)).toBe(6);
    expect(getIdealCardsLargeLimit(MAX_CARD_LIMIT, NUM_CARDS2, CARDS_PER_ROW)).toBe(6);
    expect(getIdealCardsLargeLimit(MAX_CARD_LIMIT, NUM_CARDS3, CARDS_PER_ROW)).toBe(3);
    expect(getIdealCardsLargeLimit(MAX_CARD_LIMIT, NUM_CARDS4, CARDS_PER_ROW)).toBe(9);
    expect(getIdealCardsLargeLimit(MAX_CARD_LIMIT, NUM_CARDS5, CARDS_PER_ROW)).toBe(9);
    expect(getIdealCardsLargeLimit(MAX_CARD_LIMIT, NUM_CARDS6, CARDS_PER_ROW)).toBe(3);
    expect(getIdealCardsLargeLimit(MAX_CARD_LIMIT, NUM_CARDS7, CARDS_PER_ROW)).toBe(3);
  });
});
