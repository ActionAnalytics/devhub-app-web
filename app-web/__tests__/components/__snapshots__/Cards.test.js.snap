// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cards
    cards={
      Array [
        Object {
          "description": "description",
          "owner": "bar",
          "repository": "foo",
          "resourcePath": "/",
          "resourceType": null,
          "title": "title",
        },
      ]
    }
    sourcePath=""
    topic="topic"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="TopicContainer"
        >
          <h1>
            topic
          </h1>
        </div>,
        <div
          className="LargeView"
        >
          <Toggle
            cardComponents={
              Array [
                <Default Card Component
                  author=""
                  description="description"
                  image=""
                  owner="bar"
                  repository="foo"
                  resourcePath="/"
                  resourceType={null}
                  title="title"
                />,
              ]
            }
            cardLimits={4}
          />
        </div>,
        <div
          className="MobileView"
        >
          <Toggle
            cardComponents={
              Array [
                <Default Card Component
                  author=""
                  description="description"
                  image=""
                  owner="bar"
                  repository="foo"
                  resourcePath="/"
                  resourceType={null}
                  title="title"
                />,
              ]
            }
            cardLimits={1}
          />
        </div>,
      ],
      "className": "CardsContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              topic
            </h1>,
            null,
          ],
          "className": "TopicContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "topic",
            },
            "ref": null,
            "rendered": "topic",
            "type": "h1",
          },
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Toggle
            cardComponents={
              Array [
                <Default Card Component
                  author=""
                  description="description"
                  image=""
                  owner="bar"
                  repository="foo"
                  resourcePath="/"
                  resourceType={null}
                  title="title"
                />,
              ]
            }
            cardLimits={4}
          />,
          "className": "LargeView",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cardComponents": Array [
              <Default Card Component
                author=""
                description="description"
                image=""
                owner="bar"
                repository="foo"
                resourcePath="/"
                resourceType={null}
                title="title"
              />,
            ],
            "cardLimits": 4,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Toggle
            cardComponents={
              Array [
                <Default Card Component
                  author=""
                  description="description"
                  image=""
                  owner="bar"
                  repository="foo"
                  resourcePath="/"
                  resourceType={null}
                  title="title"
                />,
              ]
            }
            cardLimits={1}
          />,
          "className": "MobileView",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cardComponents": Array [
              <Default Card Component
                author=""
                description="description"
                image=""
                owner="bar"
                repository="foo"
                resourcePath="/"
                resourceType={null}
                title="title"
              />,
            ],
            "cardLimits": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="TopicContainer"
          >
            <h1>
              topic
            </h1>
          </div>,
          <div
            className="LargeView"
          >
            <Toggle
              cardComponents={
                Array [
                  <Default Card Component
                    author=""
                    description="description"
                    image=""
                    owner="bar"
                    repository="foo"
                    resourcePath="/"
                    resourceType={null}
                    title="title"
                  />,
                ]
              }
              cardLimits={4}
            />
          </div>,
          <div
            className="MobileView"
          >
            <Toggle
              cardComponents={
                Array [
                  <Default Card Component
                    author=""
                    description="description"
                    image=""
                    owner="bar"
                    repository="foo"
                    resourcePath="/"
                    resourceType={null}
                    title="title"
                  />,
                ]
              }
              cardLimits={1}
            />
          </div>,
        ],
        "className": "CardsContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                topic
              </h1>,
              null,
            ],
            "className": "TopicContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "topic",
              },
              "ref": null,
              "rendered": "topic",
              "type": "h1",
            },
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Toggle
              cardComponents={
                Array [
                  <Default Card Component
                    author=""
                    description="description"
                    image=""
                    owner="bar"
                    repository="foo"
                    resourcePath="/"
                    resourceType={null}
                    title="title"
                  />,
                ]
              }
              cardLimits={4}
            />,
            "className": "LargeView",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "cardComponents": Array [
                <Default Card Component
                  author=""
                  description="description"
                  image=""
                  owner="bar"
                  repository="foo"
                  resourcePath="/"
                  resourceType={null}
                  title="title"
                />,
              ],
              "cardLimits": 4,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Toggle
              cardComponents={
                Array [
                  <Default Card Component
                    author=""
                    description="description"
                    image=""
                    owner="bar"
                    repository="foo"
                    resourcePath="/"
                    resourceType={null}
                    title="title"
                  />,
                ]
              }
              cardLimits={1}
            />,
            "className": "MobileView",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "cardComponents": Array [
                <Default Card Component
                  author=""
                  description="description"
                  image=""
                  owner="bar"
                  repository="foo"
                  resourcePath="/"
                  resourceType={null}
                  title="title"
                />,
              ],
              "cardLimits": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
